### PR TITLE
Implement the /register endpoint

### PR DESCRIPTION
**Pull request by SShrike**
This is some more work towards implementing the [user data endpoints](http://matrix.org/docs/api/client-server/#!/User_data).

`_dummy_user` has also been turned into an entry into a new `users` dictionary, which allows us to add accounts, etc. Now of course any users added and/or modified will be wiped on the creation of a new `BackgroundServer` or on restarting the server from the command line, but that doesn't matter as Bluepill is a *mock* server and as such is not intended to permanently store any data.

### CODE REVIEW

File: bluepill/server/resource/__init__.py [hunk 1]
```diff
@@ -13,29 +13,41 @@
...
     return {
         'errcode': code,
         'error': msg,
     }
 
+def gen_token():
+    """Generate a random ten digit fake token."""
+    length = 10
+    lower = 10**(length - 1)
+    upper = 10**length - 1
```
Comment (Half-Shot) [inline bluepill/server/resource/__init__.py]:
```
>>> 10**(10-1)
1000000000
>>> 10**10-1
9999999999
```

What?

**Comment by SShrike** [comment]
Everything should be ready to go, I've cleaned up the code, written the appropriate tests, and made sure everything worked correctly (to the best of my abilities).

**Comment by SShrike** [comment]
Just added a couple of tests for areas I missed.

One included a bit of an [ugly regex](https://github.com/SShrike/bluepill/pull/14/commits/985517ea700f65c6b934b57fd903c3743f72506f#diff-048f646cad906849b5e463291db0ef73R69) sadly, any suggestions on how I get rid of it?

**Comment by Half-Shot** [comment]
(Github should let me reply to comments >:( )

> One includes a bit of an ugly regex sadly, any suggestions for a nicer solution?

Wouldn't this be more correct and also a little nicer to read ``\A@([^@:]*):``
You either use a regex or you do a userid[1:PositionOfFirstColon].

**Pull request merged**